pftp/client_handler.go

```diff
@@ -110,30 +103,29 @@ func (c *clientHandler) handleCommands() error {
 			if err != nil {
 				if err != io.EOF {
 					safeSetChanel(proxyError, err)
+				} else {
+					proxyError <- err
 				}
 				break
 			}
 		}
-		done <- struct{}{}
+
+		if c.proxy != nil {
+			c.proxy.Close()
+		}
 	}()
 
-	for {
-		select {
-		case e := <-proxyError:
-			return e
-		default:
-			if c.config.IdleTimeout > 0 {
-				c.setClientDeadLine(c.config.IdleTimeout)
-			}
-			line, err := c.reader.ReadString('\n')
+	go func() {
```

Comment:
I think that complicated.because two goroutine in this method ,
let's name it inside groutine.

```go
# example
func readRemote() {
...
}
go readRemote()
```

Reply:
fixed
https://github.com/pyama86/pftp/pull/26/commits/1ea48e6931939a2e9a8ef47743c620d57a32a519